src/ReactPlaceholder.js

```diff
@@ -40,14 +40,20 @@ export default class ReactPlaceholder extends React.Component {
       type, customPlaceholder, showLoadingAnimation, ...rest
     } = this.props;
 
-    if (customPlaceholder) {
-      return customPlaceholder;
-    }
-
     const classes = showLoadingAnimation ?
       ['show-loading-animation', className].filter(c => c).join(' ') :
       className;
 
+    if (customPlaceholder && React.isValidElement(customPlaceholder)) {
+      const mergedCustomClasses =
+        [customPlaceholder.props.className, classes].filter(c => c).join(' ');
+      return React.cloneElement(customPlaceholder,
```

Comment:
Yes, it works. I fixed examples and test it. Test more if you want to sure.
https://github.com/buildo/react-placeholder/blob/master/examples/examples.js#L45